src/main/webapp/app/shared/refresh/refresh-selector.component.ts

```diff
@@ -0,0 +1,112 @@
+import { Component, OnInit, OnDestroy, Input } from '@angular/core';
+import { Subscription } from 'rxjs/Subscription';
+
+import { Observable } from 'rxjs/Observable';
+import { JhiRefreshService } from './refresh.service';
+
+@Component({
+    selector: 'jhi-refresh-selector',
+    templateUrl: './refresh-selector.component.html',
+    styleUrls: [
+        'refresh-selector.component.scss'
+    ]
+})
+export class JhiRefreshSelectorComponent implements OnInit, OnDestroy {
+
+    activeRefreshTime: number;
+    refreshTimes: number[];
+    refreshTimer: Subscription;
+    refreshChangedSubscription: Subscription;
+
+    /** True if dropdown-menu must be open in left direction (right direction is by default) **/
+    @Input()
+    toTheLeft: boolean = false;
```

Comment:
seems like this is only set from the directive as property and this sets a class. Why not just set the class in the directive and write a css rule like below. use `refresh-left-side` as class name instead of `left-side`
```
.refresh-left-side {
  .dropdown-menu {
    left: initial;
    right: 0;
  }
}
```
so the rule is applied on the dropdown only when the class is set on the directive